wallet/chainntfns.go

```diff
@@ -240,6 +240,29 @@ func (w *Wallet) ChainSwitch(ctx context.Context, forest *SidechainForest, chain
 
 		for _, hash := range hashes {
 			w.NtfnServer.notifyRemovedTransaction(*hash)
+
+			// Remove pruned tickets from VSP clients.
+			for _, c := range w.vspClients {
+				c.RemoveTicket(*hash, "pruned")
+			}
+
+			// Remove pruned tickets from database. First check for existence to
+			// avoid unnecessary write txns.
+			err = walletdb.View(ctx, w.db, func(tx walletdb.ReadTx) error {
```

Comment:
doing these db views and updates in a loop is a bit strange, can we do this after this for loop we're in, with only a single view and/or update?

Reply:
nvm, i think it's fine as is. if one removal fails we don't want to have to rollback the updates for the others.